pybossa/sched.py

```diff
@@ -65,24 +63,28 @@ def new_task(project_id, sched, user_id=None, user_ip=None,
         'depth_first_all': get_depth_first_all_task
     }
     scheduler = sched_map.get(sched, sched_map['default'])
-
     project = project_repo.get(project_id)
-    # This is here for testing. It removes the random variable to make testing deterministic.
     disable_gold = not project.info.get('enable_gold', True)
-    present_gold_task = False if gold_only or disable_gold else not random.randint(0, 10)
-    return scheduler(
-        project_id,
-        user_id,
-        user_ip,
-        external_uid,
-        offset=offset,
-        limit=limit,
-        orderby=orderby,
-        desc=desc,
-        rand_within_priority=rand_within_priority,
-        present_gold_task=present_gold_task,
-        gold_only=gold_only
-    )
+
+    task_type = 'gold_last'
+    if disable_gold:
+        # This is here for testing. It removes the random variable to make testing deterministic.
+        task_type = 'no_gold'
+    elif gold_only:
+        task_type = 'gold'
+    elif not random.randint(0, 10):
```

Comment:
Just curious, but I'm wondering if this could be rewritten as `elif random.randint(0,10) == 0` instead of the `not` operator?

`randint(0,10)` is `0-10` inclusive, and it appears this line intended to reference a value of `0`. Whichever format you feel is more clear. Thanks. :)

Reply:
This is just copied from it's original form.  I suppose I am ok with it.

Reply:
I agree with @primaryobjects that the intent of that code is obscure. And also that it was meant to be 1/10 tasks are gold but accidentally is 1/11. And that it should not be a hard-coded probability but should be configurable per project.

However, we should change it in a separate story and not mix it up with an unrelated issue.